### PR TITLE
Github Actions ci/cd config to test dependencies & src code automatically, as well as to deploy to npm registry

### PR DESCRIPTION
**Pull request by nemanja255**
This merge contains config changes for ci/cd

### CODE REVIEW

**Comment by nemanja255** [comment]
Commit [da6d0da](https://github.com/nebulr-group/nblocks-nestjs/pull/10/commits/da6d0da73d782da100977d01ce2638581bf8cbd0) was amended two times hence the -f flag was used. Having new commit would be redundant.

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -1,23 +1,28 @@
...
+			"extensions": [
+				"dbaeumer.vscode-eslint",
+				"kumar-harsh.graphql-for-vscode",
+				"pkief.material-icon-theme",
+				"esbenp.prettier-vscode",
+				"mikestead.dotenv",
+				"redhat.vscode-yaml"
+			],	
+			"settings": {
+				"terminal.integrated.profiles.shell.linux": "/bin/bash"
```
Comment (orr3) [inline .devcontainer/devcontainer.json:22]: This doesn't work anymore. Please help fixing this

File: .devcontainer/init.sh [hunk 2]
```diff
@@ -2,16 +2,20 @@
 ## Usage
 ## ./init_local.sh WORKSPACE_FOLDER_PATH
 PWD=$(dirname $0)
+source $PWD/secrets.env.default
```
Comment (orr3) [inline .devcontainer/init.sh:5]: Revert changes

Comment (orr3) [reply .devcontainer/init.sh:5]: But keep it here at the top

File: .github/dependabot.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+# https://docs.github.com/github/administering-a-repository/configuration-options-for-dependency-updates
+
+version: 2
+updates:
+  - package-ecosystem: "npm" # See documentation for possible values
+    directory: "/" # Location of package manifests
+    # Raise pull requests for version updates
+    # to npm against the `develop` branch,
+    # Note: When you use this option, the settings for this package manager will no longer affect any pull requests raised for security update.
+    target-branch: "master"
```
Comment (orr3) [inline .github/dependabot.yml:13]: Change branch

File: .github/workflows/ci-cd.yml [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+      - name: Step 2 - Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      # Push to Docker Engine
+      - name: Step 3 - Build Image & Export to Docker Engine
+        uses: docker/build-push-action@v2
+        with:
+          push: false
+          tags: nblocks-nestjs:latest
+          target: test
+          outputs: type=docker,dest=/tmp/nblocks-nestjs.tar
```
Comment (orr3) [inline .github/workflows/ci-cd.yml:27]: Can we have `nblocks-nestjs` as a variable and reference it everywhere?

File: publish.sh [hunk 5]
```diff
@@ -0,0 +1,2 @@
+#! /bin/sh
```
Comment (orr3) [inline publish.sh:1]: Rename file to something better. Don't forget to refactor where this file is suppose to be called from. chmod +x????

**Review by orr3 — CHANGES_REQUESTED** [review]
Nice work man!

**Pull request merged**